File: Recursion/Code/factorials.py
# Recursion
# Factorial of number n

def factorial(N):

    if (N==0 or N==1): #Base Case
        return 1
        
    else:
        #return N*factorial(N-1) #one way - recursive call & small calculation in same statement
        #other way
        recResult = factorial(N-1)  # recursive call
        result = N * recResult      # small calculation
        return result


n = int(input("Enter a number whose factorial is needed : "))

print("Factorial of ",n," is = ",factorial(n))

File: Recursion/Code/factorials.c
// Recursion
// Factorial of number n

#include <stdio.h>
long long int factorial(int n) {

    if (n==0 || n==1){ //Base Case
        return 1;
    }
    
    else{
        //return n * factorial(n-1) one way
        //other way

        int recResult = factorial(n-1); // Recursive Call
        int result = n*recResult;       // Small Calculation
        return result;
    }
}

int main(){

    int N;
    printf("Enter a number whose factorial we need : ");
    scanf("%d",&N);
    
    printf("Factorial of %d",N);
    printf(" is = %lld \n",factorial(N));
}
